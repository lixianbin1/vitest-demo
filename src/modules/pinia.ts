import { App } from 'vue'
import { createPinia } from 'pinia'

export const PiniaStore = createPinia()
export const install = (app:App) => {
  app.use(PiniaStore)
}
export default install