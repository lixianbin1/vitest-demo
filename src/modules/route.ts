import { App } from 'vue'
import { createRouter,createWebHashHistory } from 'vue-router'
import { setupLayouts } from 'virtual:generated-layouts'
import generatedRoutes from '~pages'

export const routes = setupLayouts(generatedRoutes)
export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
export const install = (app:App) => {
  app.use(router)
}
export default install