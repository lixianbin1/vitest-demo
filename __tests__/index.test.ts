import {expect,test,vi} from 'vitest'
import {render} from '@testing-library/vue'
import HomePage from '../src/pages/index.vue'
import {i18n} from '../src/modules/i18n'
import {router} from '../src/modules/route'

test('HomePage页面的快照',()=>{
  const { container } = render(HomePage,{
    global: {
      plugins:[i18n,router]
    }
  })
  expect(container).toMatchSnapshot()
})