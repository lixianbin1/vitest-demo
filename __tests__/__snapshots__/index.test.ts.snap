// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage页面的快照 1`] = `
<div>
  <div
    class="el-table--fit el-table--enable-row-hover el-table--enable-row-transition el-table el-table--layout-fixed"
    data-prefix="el"
    data-v-d1d3d0d7=""
    style="height: 100%;"
  >
    <div
      class="el-table__inner-wrapper"
    >
      <div
        class="hidden-columns"
      >
        
        <div
          data-v-d1d3d0d7=""
        />
        <div
          data-v-d1d3d0d7=""
        />
        <div
          data-v-d1d3d0d7=""
        />
        
      </div>
      <div
        class="el-table__header-wrapper"
      >
        <table
          border="0"
          cellpadding="0"
          cellspacing="0"
          class="el-table__header"
        >
          <colgroup />
          <thead
            class=""
          >
            <tr
              class=""
            />
          </thead>
        </table>
      </div>
      <div
        class="el-table__body-wrapper"
      >
        <div
          class="el-scrollbar"
        >
          <div
            class="el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
          >
            <div
              class="el-scrollbar__view"
              style="display: inline-block; vertical-align: middle;"
            >
              
              <table
                border="0"
                cellpadding="0"
                cellspacing="0"
                class="el-table__body"
                style="table-layout: fixed;"
              >
                <colgroup />
                <!--v-if-->
                <tbody
                  tabindex="-1"
                >
                  
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  <tr
                    class="el-table__row"
                  />
                  
                </tbody>
                <!--v-if-->
              </table>
              <!--v-if-->
              <!--v-if-->
              
            </div>
          </div>
          
          <transition-stub
            appear="false"
            css="true"
            name="el-scrollbar-fade"
            persisted="true"
          >
            <div
              class="el-scrollbar__bar is-horizontal"
              style="display: none;"
            >
              <div
                class="el-scrollbar__thumb"
                style="transform: translateX(0%);"
              />
            </div>
          </transition-stub>
          <transition-stub
            appear="false"
            css="true"
            name="el-scrollbar-fade"
            persisted="true"
          >
            <div
              class="el-scrollbar__bar is-vertical"
              style="display: none;"
            >
              <div
                class="el-scrollbar__thumb"
                style="transform: translateY(0%);"
              />
            </div>
          </transition-stub>
          
        </div>
      </div>
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="el-table__column-resize-proxy"
      style="display: none;"
    />
  </div>
</div>
`;
